documentation/projects/proposals/nuxt_3_migration/20230608-implementation_plan_nuxt_3_migration.md

```diff
@@ -0,0 +1,338 @@
+some components if they need extensive code changes anyway, or if the behavior
+within them changes significantly and the `script setup` simplifies the
+component. The decision whether to convert components to `script setup` would be
+at the discretion of the PR implementer, but e should not refactor for
+refactoring's sake alone, after all.
+
+#### Testing
+
+Vue 3 recommends to use Vitest for better performance and integration with
+`vite` (which will replace `webpack` in our setup). I have not had an
+opportunity to test it out, but I think it would make sense to convert our unit
+tests to `Vitest`. The decision on whether to keep using the current
+dependencies or convert to `Vitest` should be based on the ease of conversion.
+[Type tests is also interesting and may be useful in some places.](https://vitest.dev/guide/testing-types.html)
+We should add
+[`eslint-plugin-vitest`](https://github.com/veritem/eslint-plugin-vitest#credits)
+which basically a port of eslint-plugin-jest with necessary modifications.
+
+## Expected Outcomes
+
+<!-- List any succinct expected products from this implementation plan. -->
+
+Openverse frontend uses Nuxt 3, and all current features (including i18n,
+sitemaps, SEO) work.
+
+## Outlined Steps
+
+<!-- Describe the implementation step necessary for completion. -->
+
+This section describes the known steps that need to be taken to migrate to
+Nuxt 3. However, it is possible that there will be some unknown steps that will
+be discovered during the implementation.
+
+### Changes to main
+
+- Deprecation of CJS modules (the ones that use `require` and `module.exports`):
+
+  - [ ] `case` package that we use for `kebab-case`, `title` case and
+        `camelCase` function is CJS. We can either find replacements that use
+        `ESM` modules, vendor in these functions, or use a workaround that
+        `vite` suggests (instead of `const { kebab } = require "case"` use
+        `import casePkg; const { kebab } = casePkg;`)
+  - [ ] home gallery uses `require` when importing the images dynamically. There
+        is a small number of them, so we should just move them to `static`
+        folder (or `public` in Nuxt 3) and use static `import`s.
+  - [ ] #2342 Tailwind config should be converted to TypeScript.
+
+- [ ] Refactor image and audio single result pages to show the skeleton page
+      when the "image"/"audio" is `null`. This is required because the way data
+      is fetched changes, and _is_ the correct way it _should_ be set up
+      (instead of not handling the case of `image` being `null` and instead
+      relying on the error page redirect)
+- [ ] Refactor skip-to-content functionality. Nuxt 3 only supports teleports in
+      ClientOnly, so the current implementation is causing server-client
+      mismatch and rehydration. We should move to a simple HTML anchor tag and
+      an `id` on the content.
+- [ ] Refactor the functionality that is called on app load and route
+      navigation.
```

Comment:
Could you elaborate on what this entails? It feels really vague and I'm not sure what it refers to.